### PR TITLE
Simplify testcafé setup

### PR DESCRIPTION
**Pull request by NSeydoux**
This replaces the current testcafé CI setup, which is based on the
vercel deployment, by a simpler alternative which runs the demo app
locally, and tests against it.

Among other benefits, it prevents running the browser-based e2e tests
for dependabot PRs.

- [X] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by pmcb55 — APPROVED** [review]

File: .github/workflows/e2e-browser.yml [hunk 1]
```diff
@@ -1,72 +1,68 @@
...
         uses: actions/setup-node@v2.1.5
+        with:
+          node-version: ${{ matrix.node-version }}
       - name: Cache node modules
         uses: actions/cache@v2.1.5
         env:
           cache-name: cache-node-modules
         with:
-          path: e2e/browser/node_modules
+          path: node_modules
```
Comment (Vinnl) [inline .github/workflows/e2e-browser.yml:26]: Heh, good one. I wonder if maybe the pipelines could be sped up some more by also caching the `/packages/*/node_modules/`?

File: .github/workflows/e2e-browser.yml [hunk 2]
```diff
@@ -1,72 +1,68 @@
...
+      - run: npm ci
+      - run: npm run bootstrap -- --ci
+      - run: npm run build
+      - name: Prepare browser-based end-to-end tests
+        run: |
+          cd packages/browser/examples/demoClientApp
+          npm install
+          cd ../../../../
+      - name: Run browser-based end-to-end tests (Linux)
+        # TODO: Add Edge/merge with Windows setup once Edge is available on Linux:
```
Comment (Vinnl) [inline .github/workflows/e2e-browser.yml:37]: Looks like you're just running on Ubuntu, so no Windows setup to merge it with :)

```suggestion
```

File: .github/workflows/e2e-browser.yml [hunk 3]
```diff
@@ -1,72 +1,68 @@
...
-          npm test;
+          cd e2e/browser
+          npm ci
+          npm run test -- "firefox:headless,chrome:headless:userAgent='Browser-based solid-client-authn end-to-end tests running in CI. Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) $(chrome --version) Safari/537.36'" --screenshots takeOnFails=true,path=./e2e-browser-failures --retry-test-pages --hostname localhost
+        # The Node version does not influence how well our tests run in the browser,
+        # so we only need to test in one.
+        # (But I've explicitly set it to *not* run in the oldest versions,
+        # rather than running in the newest, so ensure that changing Node versions in CI
+        # does not cause end-to-end tests to stop running.)
+        # Additionally, Dependabot does not have access to our secrets,
```
Comment (Vinnl) [inline .github/workflows/e2e-browser.yml:51]:
```suggestion
        # Dependabot does not have access to our secrets,
```

**Review by Vinnl — APPROVED** [review]
"Oh look at me, I'm Nicolas and I'm French, and I can easily type an accent aigu."

Much better this though :)

File: .github/workflows/e2e-browser.yml (hunk 1 above)
Comment (NSeydoux) [reply .github/workflows/e2e-browser.yml:26]: That's an excellent question. I'll give it a try and see, the action documentation seems to lean towards this.

**Pull request merged**